File: 2021/7/part_2.py
with open("input.txt", "r") as f:
    lines = [l.rstrip("\n") for l in f.readlines()]

positions =[*map(int, lines[0].split(","))]
p_min = min(positions)
p_max = max(positions)

fuel_usages = {}

for i in range(p_min, p_max+1):
    fuel = 0
    for p in positions:
        steps = abs(i - p)
        fuel += (steps**2)//2+(steps-(steps//2))

    fuel_usages[i] = fuel

print(min(fuel_usages.values()))

File: 2021/7/part_2.nim
import std/[strutils, sugar]

let
    data = readFile("input.txt")
    lines = data.strip(true, true, {'\n', '\r', ' '}).splitLines()

    positions = collect(newSeq):
        for l in lines[0].split({','}):
            l.parseInt

    pMin = positions.min
    pMax = positions.max

var fuelUsages = newSeq[int]()

for i in pMin .. pMax:
    var fuel = 0

    for p in positions:
        let steps = abs(i - p)
        fuel += ((steps * (steps + 1)) / 2).toInt

    fuelUsages.add(fuel)

echo min(fuelUsages)
